compiler/rustc_lint/src/types.rs

```diff
@@ -770,8 +773,26 @@ pub(crate) fn repr_nullable_ptr<'tcx>(
     debug!("is_repr_nullable_ptr(cx, ty = {:?})", ty);
```

Comment:
Also, the doc comment seems to suggest we have similar code in codegen, but I don't see it changed in this PR.